ArcGISRuntimeSDKQt_CppSamples/Geometry/ConvexHull/ConvexHull.cpp

```diff
@@ -0,0 +1,135 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "ConvexHull.h"
+
+#include "GeometryEngine.h"
+#include "Graphic.h"
+#include "GraphicsOverlay.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MultipointBuilder.h"
+#include "PointCollection.h"
+#include "SimpleFillSymbol.h"
+#include "SimpleMarkerSymbol.h"
+
+using namespace Esri::ArcGISRuntime;
+
+ConvexHull::ConvexHull(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(Basemap::topographic(this), this))
+{
+  setupGraphics();
+}
+
+ConvexHull::~ConvexHull() = default;
+
+void ConvexHull::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ConvexHull>("Esri.Samples", 1, 0, "ConvexHullSample");
+}
+
+MapQuickView* ConvexHull::mapView() const
+{
+  return m_mapView;
+}
+
+void ConvexHull::displayConvexHull()
+{
+  Geometry convexHull = GeometryEngine::convexHull(m_inputsGraphic->geometry());
+
+  // change the symbol based on the returned geometry type
+  if (convexHull.geometryType() == GeometryType::Point)
+  {
+    m_convexHullGraphic->setSymbol(m_markerSymbol);
+  }
+  else if (convexHull.geometryType() == GeometryType::Polyline)
+  {
+    m_convexHullGraphic->setSymbol(m_lineSymbol);
+  }
+  else if (convexHull.geometryType() == GeometryType::Polygon)
+  {
+    m_convexHullGraphic->setSymbol(m_fillSymbol);
+  }
+
+  m_convexHullGraphic->setGeometry(convexHull);
+  return;
+}
+
+void ConvexHull::clearGraphics()
+{
+  m_inputs.clear();
+  m_inputsGraphic->setGeometry(Geometry());
+  m_convexHullGraphic->setGeometry(Geometry());
+}
+
+void ConvexHull::setupGraphics()
+{
+  // graphics overlay to show clicked points and convex hull
+  m_graphicsOverlay = new GraphicsOverlay(this);
+
+  // create a graphic to show clicked points
+  m_markerSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle, Qt::red, 10, this);
+  m_inputsGraphic = new Graphic(this);
+  m_inputsGraphic->setSymbol(m_markerSymbol);
+  m_graphicsOverlay->graphics()->append(m_inputsGraphic);
+
+  // create a graphic to display the convex hull
+  m_convexHullGraphic = new Graphic(this);
+  m_graphicsOverlay->graphics()->append(m_convexHullGraphic);
+
+  // create a graphic to show the convex hull
+  m_lineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, Qt::blue, 3, this);
+  m_fillSymbol = new SimpleFillSymbol(SimpleFillSymbolStyle::Null, Qt::transparent, m_lineSymbol, this);
+}
+
+void ConvexHull::getInputs()
+{
+  // show clicked points on MapView
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& e){
+    e.accept();
+
+    const Point clickedPoint = m_mapView->screenToLocation(e.x(), e.y());
+    m_inputs.push_back(clickedPoint);
+
+    PointCollection* pointCollection = new PointCollection(m_mapView->spatialReference(), this);
```

Comment:
This isn't a memory leak, but it is a space leak.

To clarify a misconception here: a `MultipointBuilder` is a _mutable_ type that creates immutable geometries. As a `Multipoint` builder fill the mutable roll - there should be no need to `new` up a new `MultipointBuilder` every time you add a new point to your list. Your `MultipointBuilder` **is** your list.

So two changes:
1. Remove `m_inputs`.
2. Create a new member variable called: `m_multipointBuilder`.

Then do the following:

```c++
 connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& e)
{
  e.accept();
  multipointBuilder->points()->addPoint(m_mapView->screenToLocation(e.x(), e.y()));
  m_inputsGraphic->setGeometry(multipointBuilder->toGeometry());
}
```